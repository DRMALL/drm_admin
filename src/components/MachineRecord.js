

import React from 'react'
import getInputValue from '../actions/getInputValue'
// import { recordKind } from '../commons/machineLable'
import Select from 'antd/lib/select'
import Input from 'antd/lib/input'
import selectValue from '../utils/selectValue'
import DatePicker from 'antd/lib/date-picker'
import dispatch from '../actions/dispatch'



export default props => {
  const { line_time, line_type, line_des, TimelineKindArr } = props
  return (
    <div onChange={affixed => console.log(affixed)} >
      <div className='machine-input-flex' >
        <div className='machine-input-left' >时间</div>
        <DatePicker onChange={ getTime } style={{ width: 400 }}
                    value={ line_time } />
      </div>
      <div className='machine-input-flex' >
          <div  className='machine-input-left'>分类</div>
          <Select style={{ width: 400 }} placeholder='选择分类'
                  onChange={ selectValue } value={ line_type } >
                  {
                    TimelineKindArr.map((item, index) =>
                      <Select.Option key={ index } value={ item.text+'/line_type' } >{ item.text }</Select.Option>)
                  }
          </Select>
        </div>
        <div className='machine-input-flexs' >
        <div className='machine-input-left' >描述</div>
        <Input.TextArea rows={4} className='machine-input-right'
                  onChange={ getInputValue }
                  style={{ width: 400 }} placeholder='输入描述'
                  data-path='MACHINE' data-id='line_des' value={ line_des } />
      </div>
    </div>
    )
}

function getTime(ment){
  dispatch('MACHINE_SELCET_TIME', ment)
}

