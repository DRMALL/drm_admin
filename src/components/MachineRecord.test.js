

import React from 'react'
import renderer from 'react-test-renderer'
import MachineRecord from './MachineRecord'


it('MachineRecord test', () => {
  // const times =
  const line_type = '安装'
  const line_des = '第一次安装'
  const tree = renderer.create(<MachineRecord  line_type={ line_type }  line_des= { line_des } />)
  expect(tree).toMatchSnapshot()
})
