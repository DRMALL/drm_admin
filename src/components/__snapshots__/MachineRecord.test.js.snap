// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MachineRecord test 1`] = `
<div
  onChange={[Function]}
>
  <div
    className="machine-input-flex"
  >
    <div
      className="machine-input-left"
    >
      时间
    </div>
    <span
      className="ant-calendar-picker"
      style={
        Object {
          "width": 400,
        }
      }
    >
      <div
        onBlur={undefined}
        onClick={[Function]}
        onFocus={undefined}
        onKeyDown={[Function]}
        onMouseDown={[Function]}
        onMouseEnter={undefined}
        onMouseLeave={undefined}
        onTouchStart={[Function]}
      >
        <input
          className="ant-calendar-picker-input ant-input"
          disabled={undefined}
          placeholder="请选择日期"
          readOnly={true}
          value=""
        />
        <span
          className="ant-calendar-picker-icon"
        />
      </div>
    </span>
  </div>
  <div
    className="machine-input-flex"
  >
    <div
      className="machine-input-left"
    >
      分类
    </div>
    <div
      className="ant-select ant-select-enabled"
      onBlur={[Function]}
      onClick={[Function]}
      onFocus={[Function]}
      onMouseDown={[Function]}
      onMouseEnter={undefined}
      onMouseLeave={undefined}
      onTouchStart={[Function]}
      style={
        Object {
          "width": 400,
        }
      }
    >
      <div
        aria-autocomplete="list"
        aria-expanded={false}
        aria-haspopup="true"
        className="ant-select-selection
                  ant-select-selection--single"
        onKeyDown={[Function]}
        role="combobox"
        tabIndex={0}
      >
        <div
          className="ant-select-selection__rendered"
        >
          <div
            className="ant-select-selection__placeholder"
            onClick={[Function]}
            onMouseDown={[Function]}
            style={
              Object {
                "WebkitUserSelect": "none",
                "display": "none",
                "userSelect": "none",
              }
            }
            unselectable="unselectable"
          >
            选择分类
          </div>
          <div
            className="ant-select-selection-selected-value"
            style={
              Object {
                "display": "block",
                "opacity": 1,
              }
            }
            title="安装"
          >
            安装
          </div>
        </div>
        <span
          className="ant-select-arrow"
          onClick={[Function]}
          style={
            Object {
              "WebkitUserSelect": "none",
              "userSelect": "none",
            }
          }
          unselectable="unselectable"
        >
          <b />
        </span>
      </div>
    </div>
  </div>
  <div
    className="machine-input-flexs"
  >
    <div
      className="machine-input-left"
    >
      描述
    </div>
    <textarea
      className="ant-input machine-input-right"
      data-id="line_des"
      data-path="MACHINE"
      onChange={[Function]}
      onKeyDown={[Function]}
      placeholder="输入描述"
      rows={4}
      style={
        Object {
          "width": 400,
        }
      }
      value="第一次安装"
    />
  </div>
</div>
`;
